nexus/src/internal_api/http_entrypoints.rs

```diff
@@ -466,6 +467,49 @@ async fn cpapi_producers_post(
         .await
 }
 
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    serde::Deserialize,
+    schemars::JsonSchema,
+    serde::Serialize,
+)]
+pub struct CollectorIdPathParams {
+    /// The ID of the oximeter collector.
+    pub collector_id: Uuid,
+}
+
+/// List all metric producers assigned to an oximeter collector.
+#[endpoint {
+     method = GET,
+     path = "/metrics/collectors/{collector_id}/producers",
+ }]
+async fn cpapi_assigned_producers_list(
+    request_context: RequestContext<Arc<ServerContext>>,
+    path_params: Path<CollectorIdPathParams>,
+    query_params: Query<PaginatedById>,
+) -> Result<HttpResponseOk<ResultsPage<ProducerEndpoint>>, HttpError> {
+    let context = request_context.context();
+    let handler = async {
+        let nexus = &context.nexus;
+        let collector_id = path_params.into_inner().collector_id;
+        let query = query_params.into_inner();
+        let pagparams = data_page_params_for(&request_context, &query)?;
+        let producers =
+            nexus.list_assigned_producers(collector_id, &pagparams).await?;
```

Comment:
Good question. I think we can do that, yeah. I explicitly ticked the periodic timer once at the start of the new refresh loop, specifically to avoid hitting Nexus right after we register. But I think that's probably not needed, and we can just do that both at startup and periodically.

Let me look into doing it now, and see if it's too disruptive to defer.

Reply:
Ok, this was very easy. I'm testing it out now locally, by killing oximeter after startup and verifying that it still re-fetches the list, but I don't foresee any issues. Thanks!

Reply:
Done in 633eb2862